pkg/controller/installation/core_controller.go

```diff
@@ -742,26 +748,41 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	terminating := (instance.DeletionTimestamp != nil)
 	preDefaultPatchFrom := client.MergeFrom(instance.DeepCopy())
 
+	// Changes for updating installation status conditions
+	if request.Name == "calico" && request.Namespace == "" {
+		ts := &operator.TigeraStatus{ObjectMeta: metav1.ObjectMeta{Name: "calico"}}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: "calico"}, ts)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.updateInstallationStatus(instance, ts.Status.Conditions)
+		if err := r.client.Status().Update(ctx, instance); err != nil {
+			log.WithValues("reason", err).Info("Failed to create Installation status conditions.")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
 	// Mark CR found so we can report converter problems via tigerastatus
 	r.status.OnCRFound()
 
 	if !r.migrationChecked {
 		// update Installation resource with existing install if it exists.
 		nc, err := convert.NeedsConversion(ctx, r.client)
 		if err != nil {
-			r.SetDegraded("Error checking for existing installation", err, reqLogger)
+			r.SetDegraded(operator.ResourceUpdateError, "Error checking for existing installation", err, reqLogger)
 			return reconcile.Result{}, err
 		}
 		if nc {
 			install, err := convert.Convert(ctx, r.client)
 			if err != nil {
 				if errors.As(err, &convert.ErrIncompatibleCluster{}) {
-					r.SetDegraded("Existing Calico installation can not be managed by Tigera Operator as it is configured in a way that Operator does not currently support. Please update your existing Calico install config", err, reqLogger)
+					r.SetDegraded(operator.ResourceValidationError, "Existing Calico installation can not be managed by Tigera Operator as it is configured in a way that Operator does not currently support. Please update your existing Calico install config", err, reqLogger)
 					// We should always requeue a convert problem. Don't return error
 					// to make sure we never back off retrying.
 					return reconcile.Result{RequeueAfter: 15 * time.Second}, nil
 				}
-				r.SetDegraded("Error converting existing installation", err, reqLogger)
+				r.SetDegraded(operator.ResourceUpdateError, "Error converting existing installation", err, reqLogger)
```

Comment:
This method does not update a resource, so I do not think we should use ResourceUpdateError here.

Reply:
Done